pandas/tests/indexes/test_common.py

```diff
@@ -391,3 +398,40 @@ def test_astype_preserves_name(self, index, dtype):
             assert result.names == index.names
         else:
             assert result.name == index.name
+
+
+@pytest.mark.parametrize("na_position", [None, "middle"])
+def test_sort_values_invalid_na_position(index_with_missing, na_position):
+
+    if type(index_with_missing) in [DatetimeIndex, PeriodIndex, TimedeltaIndex]:
+        pytest.xfail("stable descending order sort not implemented")
+    elif type(index_with_missing) in [CategoricalIndex, MultiIndex]:
+        pytest.xfail("missing value sorting order not defined for index type")
+
+    if na_position not in ["first", "last"]:
+        with pytest.raises(
+            ValueError, match=f"invalid na_position: {na_position}",
+        ):
+            index_with_missing.sort_values(na_position=na_position)
+
+
+@pytest.mark.parametrize("na_position", ["first", "last"])
+def test_sort_values_with_missing(index_with_missing, na_position):
+    # GH 35584. Test that sort_values works with missing values,
+    # sort non-missing and place missing according to na_position
+
+    if type(index_with_missing) in [DatetimeIndex, PeriodIndex, TimedeltaIndex]:
```

Comment:
same

Reply:
@jreback 
Sorry, I wasn't at all clear in the xfail messages as to how the tests are different. The one in `period/test_ops.py` checks whether `PeriodIndex` and another non-datetime-like subtype sort duplicates into the same order when `descending=True`. As part of syncing duplicate sorting behavior in the next PR, I'd like to write a general test with an `index_with_duplicates` fixture to check for sorting stability and replace this test.

The tests in `test_common.py` xfail, because the `na_position` kwarg isn't implemented yet for datetime-like indices. Implementing it would go with implementing stable duplicate sorting behavior and break a bunch of tests. I also intend to deal with that in the next PR as I bring sort_values in sync between all Index subtypes and Series.

I've clarified the xfail messages everywhere and added comments with xrefs to the tests in `test_common.py`.  Is that acceptable? I think keeping the tests separate makes sense for now.

Sorry if I misunderstood your question.